### PR TITLE
[tagger] change prefix for container runtimes and k8s pods

### PR DESCRIPTION
**Pull request by hkaj**
### What does this PR do?

Removes the runtime name for a container and simply
use "container_id://". As for pods, kubernetes_pod:// becomes
kubernetes_pod_uid://.

### Motivation

The goal of this change is to make it simpler
to use the tagger from other agents (trace in this case) w/o having to
determine the runtime in use.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/util/kubernetes/kubelet/containers.go [hunk 1]
```diff
@@ -185,3 +190,15 @@ func parseContainerReadiness(status ContainerStatus, pod *Pod) string {
 	}
 	return containers.ContainerUnhealthy
 }
+
+// KubeContainerIDToEntityID builds an entity ID from a container ID coming from
+// the pod status (i.e. including the <runtime>:// prefix).
+func KubeContainerIDToEntityID(ctrID string) (string, error) {
+	if strings.Contains(ctrID, "://") {
```
Comment (gbbr) [inline pkg/util/kubernetes/kubelet/containers.go]: You could simplify here, in the same style like you did elsewhere:
```go
if !strings.Contains(ctrID, "://") {
    return "", fmt.Errorf("can't extract an entity ID from container ID %s", ctrID)
}
// ...
```

File: pkg/tagger/collectors/ecs_fargate_extract_test.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ func TestParseMetadata(t *testing.T) {
 		},
 		{
 			Source: "ecs_fargate",
-			Entity: "docker://0fc5bb7a1b29adc30997eabae1415a98fe85591eb7432c23349703a53aa43280",
+			Entity: "container_id://0fc5bb7a1b29adc30997eabae1415a98fe85591eb7432c23349703a53aa43280",
```
Comment (gbbr) [inline pkg/tagger/collectors/ecs_fargate_extract_test.go]: I see `docker` was replaced everywhere, but I don't see `kubernetes://` being replaced. I think this package is a bit out of my area of expertise. Happy to try it out as soon as it gets merged. Or, if you wanna try it out from the branch, we can do that too (it's code freeze soon).

File: pkg/util/kubernetes/kubelet/containers.go (hunk 1 above)
Comment (mfpierre) [reply pkg/util/kubernetes/kubelet/containers.go]: In fact you can even simplify further by doing directly

```go
strings.LastIndexAny(s, "://")
```

which allows to remove the `Contains` and is safer if somehow the container id contains a `/` 👍

File: pkg/util/kubernetes/kubelet/container_test.go [hunk 3]
```diff
@@ -159,3 +160,20 @@ func (suite *ContainersTestSuite) TestParseContainerReadiness() {
...
+
+func TestKubeContainerIDToEntityID(t *testing.T) {
+	for in, out := range map[string]string{
+		"container_id://deadbeef": "container_id://deadbeef",
+		"containerd://deadbeef":   "container_id://deadbeef",
+		"cri-o://deadbeef":        "container_id://deadbeef",
+		"cri-o://d":               "container_id://d",
+		"container_id://":         "",
+		"deadbeef":                "",
+		"/deadbeef":               "",
```
Comment (mfpierre) [inline pkg/util/kubernetes/kubelet/container_test.go]: We should add a test `runtime://foo/bar` to make it explicit that we expect `container_id://foo/bar`

File: pkg/tagger/collectors/ecs_fargate_extract_test.go (hunk 2 above)
Comment (hkaj) [reply pkg/tagger/collectors/ecs_fargate_extract_test.go]: we don't replace `kubernetes`, but `kubernetes_pod://`

File: pkg/util/kubernetes/kubelet/containers.go (hunk 1 above)
Comment (hkaj) [reply pkg/util/kubernetes/kubelet/containers.go]: ❤️

**Review by mfpierre — APPROVED** [review]

**Pull request merged**